### PR TITLE
Feat/moremvc

### PR DESCRIPTION
**Pull request by ShizuokaTerrier**
All endpoints tested and working.

### CODE REVIEW

File: script.ts [hunk 1]
```diff
@@ -1,99 +1,108 @@
...
+import { PrismaClient } from '@prisma/client';
+const prisma = new PrismaClient();
 
 async function main() {
-
-  await prisma.job_table.deleteMany() // delete the existing data everytime
-  await prisma.department_table.deleteMany() // delete the existing data everytime
-  await prisma.location_table.deleteMany() // delete the existing data everytime
-  await prisma.hr_employees_table.deleteMany() // delete the existing data everytime
+  // await prisma.job_table.deleteMany() // delete the existing data everytime
```
Comment (RH-Kevin) [inline script.ts:5]: Please remove commented out code.

File: script.ts [hunk 2]
```diff
@@ -1,99 +1,108 @@
...
-        first_name: "Bob",
-        last_name: "Smith",
-        email: "bob.smith@example.com",
-        phone_number: "123123464",
+        first_name: 'Bob',
+        last_name: 'Smith',
+        email: 'bob.smith@example.com',
+        phone_number: '123123464',
         job_id: 2,
         manager_id: null, // TODO: this is not working
```
Comment (RH-Kevin) [inline script.ts:43]: Is this working now?

File: script.ts [hunk 3]
```diff
@@ -1,99 +1,108 @@
...
-    })
+  // const departments = await prisma.department_table.createMany({
+  //   data: [
+  //     { department_name: 'Engineering' },
+  //     { department_name: 'Human Resources' },
+  //     { department_name: 'Marketing' },
+  //     { department_name: 'Sales' },
+  //     { department_name: 'Customer Support' },
+  //   ],
+  // });
```
Comment (RH-Kevin) [inline script.ts:60]: Careful with leaving in the commented out code.

File: src/departments/departments.controller.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    res.status(500).json({ message: err.message });
+    console.error(err.message);
+  }
+};
+
+export const getSingleDepartment = async (req: Request, res: Response) => {
+  try {
+    const id = parseInt(req.params.id);
+    if (!id) {
+      res.status(400).json({ message: 'Invalid departmentID.' });
```
Comment (RH-Kevin) [inline src/departments/departments.controller.ts:21]: Change to "department ID"

File: src/index.ts [hunk 5]
```diff
@@ -1,29 +1,48 @@
...
+import cors from 'cors';
+import { Request, Response, Router } from 'express';
+import dotenv from 'dotenv';
+import * as EmployeeController from './hr_employees/hr_employees.controller';
+import * as DepartmentController from './departments/departments.controller';
+import * as JobsController from './jobs/jobs.controller';
+import * as LocationController from './locations/locations_controller';
+const express = require('express');
+const app = express();
+const router = Router(); // TODO: What is used for.
```
Comment (RH-Kevin) [inline src/index.ts:10]: Is this complete?

File: src/index.ts [hunk 6]
```diff
@@ -1,29 +1,48 @@
...
-})
- 
-export default router
-
-app.listen( process.env.PORT || 8080, () => {
-  console.log("server is running on port 8080 :)")
-})
\ No newline at end of file
+app.listen(process.env.PORT || 8080, () => {
+  console.log('server is running on port 8080 :)');
```
Comment (RH-Kevin) [inline src/index.ts:47]: Please change to "Server"

File: src/jobs/jobs.controller.ts [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+    res.status(500).json({ message: err.message });
+  }
+};
+
+export const deleteSingleJob = async (req: Request, res: Response) => {
+  try {
+    const id = parseInt(req.params.id);
+    const deleteJob = await JobsModel.deleteJob(id);
+    if (!deleteJob) {
+      res.status(400).json({ message: 'That job ID does not exist' });
```
Comment (RH-Kevin) [inline src/jobs/jobs.controller.ts:51]: Please change to " `Job ${id} not found in database.` "

File: src/locations/locations_controller.ts [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+    res.status(500).json({ message: err.message });
+  }
+};
+
+export const deleteSingleLocation = async (req: Request, res: Response) => {
+  try {
+    const id = parseInt(req.params.id);
+    const deleteLocation = await LocationModel.deleteLocation(id);
+    if (!deleteLocation) {
+      res.status(400).json({ message: 'That location ID does not exist' });
```
Comment (RH-Kevin) [inline src/locations/locations_controller.ts:51]: Please change to " Job ${id} not found in database. "

Comment (ShizuokaTerrier) [reply src/locations/locations_controller.ts]: Done.

File: src/jobs/jobs.controller.ts (hunk 7 above)
Comment (ShizuokaTerrier) [reply src/jobs/jobs.controller.ts]: Done.

File: src/index.ts (hunk 6 above)
Comment (ShizuokaTerrier) [reply src/index.ts]: Done.

File: src/index.ts (hunk 5 above)
Comment (ShizuokaTerrier) [reply src/index.ts:10]: Yes.

File: src/departments/departments.controller.ts (hunk 4 above)
Comment (ShizuokaTerrier) [reply src/departments/departments.controller.ts]: Done.

File: script.ts (hunk 1 above)
Comment (ShizuokaTerrier) [reply script.ts:5]: We have a reason for this. We can loop you in on site and fix it later.

File: script.ts (hunk 2 above)
Comment (ShizuokaTerrier) [reply script.ts:43]: We have a reason for this. We can loop you in on site and fix it later.

File: script.ts (hunk 3 above)
Comment (ShizuokaTerrier) [reply script.ts:60]: We have a reason for this. We can loop you in on site and fix it later.

File: script.ts (hunk 1 above)
Comment (RH-Kevin) [reply script.ts:5]: Gotcha!

File: script.ts (hunk 2 above)
Comment (RH-Kevin) [reply script.ts:43]: Got it!

**Pull request merged**